src/mme-app/actionHandlers/attachActionHandlers.cpp

```diff
@@ -646,6 +649,11 @@ ActStatus ActionHandlers::process_esm_info_resp(SM::ControlBlock& cb)
 
 	const s1_incoming_msg_data_t* s1_msg_data = static_cast<const s1_incoming_msg_data_t*>(msgBuf->getDataPointer());
 	const struct esm_resp_Q_msg &esm_res =s1_msg_data->msg_data.esm_resp_Q_msg_m;
+    
+    	if (esm_res.status != SUCCESS)
+    	{
+        	log_msg(LOG_ERROR, "ESM Response failed \n");
+    	}
 
 	procedure_p->setRequestedApn(Apn_name(esm_res.apn));
```

Comment:
if ESM response failed then we should not use esm_res.apn.  You may wish to open a enhancement defect in MME to handle ESM response failure.

Reply:
Opened EDGEPOD-420